src/common/Fixed.ml

```diff
@@ -0,0 +1,236 @@
+(** This module defines the signatures and `Make` functors for the 'fixed point'
+    (or two-level) type we use for our intermediate representations
+*)
+open Core_kernel
+
+(** The fixed-point of `Pattern.t` annotated with some meta-data *)
+module type S = sig
+  module Pattern : Pattern.S
+
+  type 'a t = {pattern: 'a t Pattern.t; meta: 'a}
+  [@@deriving compare, map, fold, hash, sexp]
+
+  val fix : 'a * 'a t Pattern.t -> 'a t
+  val unfix : 'a t -> 'a * 'a t Pattern.t
```

Comment:
Do we need this?

Reply:
As with the `Meta` helpers, I would much prefer to hide the implementation of the `Fixed.t` so both are exposed. From my point of view, they are much lighter weight than record syntax.